sqlmesh/utils/dag.py

```diff
@@ -93,6 +95,9 @@ def sorted(self) -> t.List[T]:
             while unprocessed_nodes:
                 next_nodes = {node for node, deps in unprocessed_nodes.items() if not deps}
 
+                if not next_nodes:
+                    raise SQLMeshError("Failed to topologically sort a DAG.")
```

Comment:
maybe a better error message like

"Detected a cycle in the DAG. Please check for cyclical references"

Reply:
I'm not sure if this is related to cycles, though. The issue at hand is related to Snowflake and my guess is that `deps` doesn't get updated in `deps -= next_nodes` just below due to inconsistencies in the name normalization.

I'll investigate a bit more and update this thread.

Reply:
Updated the message to hint about circular references after all.